internal/liqonet/network-manager/netcfgcreator/networkconfig.go

```diff
@@ -35,11 +35,19 @@ import (
 	foreignclusterutils "github.com/liqotech/liqo/pkg/utils/foreignCluster"
 )
 
-// GetLocalNetworkConfig returns the local NetworkConfig associated with a given cluster ID.
+// GetLocalNetworkConfig returns the local NetworkConfig associated with a given label selector and a clusterID.
 // In case more than one NetworkConfig is found, all but the oldest are deleted.
-func GetLocalNetworkConfig(ctx context.Context, c client.Client, clusterID, namespace string) (*netv1alpha1.NetworkConfig, error) {
+func GetLocalNetworkConfig(ctx context.Context, c client.Client, labels client.MatchingLabels,
+	clusterID, namespace string) (*netv1alpha1.NetworkConfig, error) {
 	networkConfigList := &netv1alpha1.NetworkConfigList{}
-	labels := client.MatchingLabels{consts.ReplicationDestinationLabel: clusterID}
+
+	if labels == nil {
+		labels = client.MatchingLabels{
+			consts.ReplicationDestinationLabel: clusterID,
+		}
+	} else {
+		labels[consts.ReplicationDestinationLabel] = clusterID
+	}
```

Comment:
Could use the `labels.Merge` function (or equivalent) if possible, which is a bit more idiomatic.

Reply:
I don't know a `labels.Merge` equivalent that works with the `client.MatchingLabels`. By using `labels.Merge` would become cumbersome to switch back and forth from the different `implementations` of labels. Any idea?